packages/core/src/shared/telemetry/util.ts

```diff
@@ -43,6 +44,20 @@ export function convertLegacy(value: unknown): boolean {
         throw new TypeError(`Unknown telemetry setting: ${value}`)
     }
 }
+/**
+ * Generate a unique identifier for the user in the form of a UUID.
+ * This identifier is unique to to the machine id and OS user name.
+ */
+export function generateTelemetryClientId(): string {
+    const userId = `${env.machineId}-${os.userInfo({ encoding: 'utf-8' }).username}`
+    const shasum = crypto.createHash('sha256')
+    const mid = shasum.update(userId).digest('hex')
+    const uid = `${mid.substring(0, 8)}-${mid.substring(8, 12)}-${mid.substring(12, 16)}-${mid.substring(
+        16,
+        20
+    )}-${mid.substring(20, 32)}`
+    return uid
```

Comment:
where did this logic come from? why are we changing it?

`globals.machineId` should be used, assuming we actually need this logic. 
https://github.com/aws/aws-toolkit-vscode/blob/3d4c65e158ff2a5a92586bad96e5229df90191ce/packages/core/src/extensionShared.ts#L98

Reply:
>where did this logic come from?

This is a unique `machineId` used by VS Code telemetry. This is not ` globals.machineId`.  These are completely different thing.

Reply:
By VS Code API definition: 
		```
/**
		 * A unique identifier for the computer.
		 */
		export const machineId: string;
```